src/main/java/io/dockstore/provision/DOSPluginUtil.java

```diff
@@ -0,0 +1,105 @@
+package io.dockstore.provision;
+
+import com.google.common.collect.Lists;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+
+class DOSPluginUtil {
+
+    private static final String API = "/ga4gh/dos/v1/dataobjects/";
+    private static final int LIST_SIZE = 3;
+    private static final int HOST = 1;
+    private static final int UID = 2;
+
+    // Package-private constructor
+    DOSPluginUtil() {
+    }
+
+    /**
+     * Gets the plugins json file path from the config file, otherwise defaults.
+     *
+     * @param dosURI The string targetPath
+     * @return The targetPath split into an ArrayList object. Return an empty List object otherwise
+     */
+    List<String> splitUri(String dosURI) {
+        if (Pattern.compile(":\\/\\/(.+)/").matcher(dosURI).find()){
+            return Lists.newArrayList(dosURI.split(":\\/\\/|/"));
+        }
+        return Collections.emptyList();
+    }
+
+    /**
+     * Gets the json response from host using HTTP GET method
+     *
+     * @param uriList The targetPath split into a List object with the following format: [scheme, host, uid]
+     * @return The JSONObject containing the content of the JSON response. Null, otherwise
+     */
+    JSONObject grabJSON(List<String> uriList){
```

Comment:
Optionally, you could have used Optional instead of returning null. :) I've just recently started using it, so now I try to (over)use it everywhere.